### DISCUSSION TITLE
any recommended links on how I can create a chat for chat completion?

### DISCUSSION THREAD

**Discussion by markat1** [withcatai/node-llama-cpp#300] [Q&A] [answered]
Would love to hear if you guys have some great (best practice) links on how to create a chat  and also make use for chat completion streaming

**Comment by giladgd** [comment, marked as answer]
You can see an example of how to do that using the stable version [here](https://withcatai.github.io/node-llama-cpp/guide/chat-session#response-streaming), and using the version 3 beta [here](https://github.com/withcatai/node-llama-cpp/pull/105#:~:text=How%20to%20stream%20a%20response)
